pkg/virt-handler/network/cniproxy/cniproxy.go

```diff
@@ -0,0 +1,150 @@
+package cniproxy
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/containernetworking/cni/libcni"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/golang/glog"
+
+	"kubevirt.io/kubevirt/pkg/virt-handler/utils"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+//TODO: Make it configurable
+const (
+	CNINetDir     = "/etc/cni/net.d"
+	CNIPluginsDir = "/opt/cni/bin"
+	LibvirtSocket = "/var/run/libvirt/libvirt-sock"
+)
+
+type CNIProxy struct {
+	cniConfig   *libcni.CNIConfig
+	netConfig   *libcni.NetworkConfig
+	runtimeConf *libcni.RuntimeConf
+}
+
+func confFiles(dir string, extensions []string) ([]string, error) {
+	// In part, adapted from rkt/networking/podenv.go#listFiles
+	files, err := ioutil.ReadDir(dir)
+
+	switch {
+	case err == nil: // break
+		glog.Errorln("err==nil")
+	case os.IsNotExist(err):
+		glog.Errorln("Doesnot exist: ", err)
+		return nil, nil
+	default:
+		glog.Errorln("default: ", err)
+		return nil, err
+	}
+
+	confFiles := []string{}
+	for _, f := range files {
+		if f.IsDir() {
+			continue
+		}
+
+		fileExt := filepath.Ext(f.Name())
+		for _, ext := range extensions {
+			glog.Errorln("match ext: ", ext)
+			if fileExt == ext {
+				glog.Errorln("adding file")
+				confFiles = append(confFiles, filepath.Join(dir, f.Name()))
+			}
+		}
+	}
+	return confFiles, nil
+}
+
+func getCNINetworkConfig() (*libcni.NetworkConfig, error) {
+	files, err := libcni.ConfFiles(CNINetDir, []string{".conf"})
+
+	switch {
+	case err != nil:
+		return nil, err
+	case len(files) == 0:
+		files, _ := ioutil.ReadDir("/etc/cni/net.d/")
+		for _, f := range files {
+			glog.Errorln(f.Name())
+		}
+		return nil, fmt.Errorf("No networks found in %s", CNINetDir)
+	}
+
+	sort.Strings(files)
+	for _, confFile := range files {
+		conf, err := libcni.ConfFromFile(confFile)
+		if err != nil {
+			glog.Warningf("Error loading CNI config file %s: %v", confFile, err)
+			continue
+		}
+		return conf, nil
+	}
+	return nil, fmt.Errorf("No valid networks found in %s", CNINetDir)
```

Comment:
Kubernetes convention: error messages in error objects are lower case (so they can be chained and still look sensible.)

Reply:
Got it, will fix :)